### ISSUE TITLE
inner product fails with (const and standard) TensorMap

### ISSUE THREAD

**Issue by matthiasneuner** [romeric/Fastor#115]
Hello,

this MWE does not compile on g++ 10 for `TensorMap` (`const` on non const):
```
    using namespace Fastor;
    double data_a[] = {1, 2, 3};
    double data_b[] = {1, 2, 3};

    const auto a = Tensor           <double, 3> (data_a);

    const auto b = Tensor           <double, 3> (data_b); 
    const auto c = TensorMap  <const double, 3> (data_b); 
    const auto d = TensorMap        <double, 3> (data_b); 

    const double res_b = inner ( a, b); // OK
    const double res_c = inner ( a, c); // fails
    const double res_d = inner ( a, d); // fails

```

Thank you again for you great work and quick response; Fastor has doubled(!) already the speed of my FE-implementation compared to the C-tran version (and considerably improved readability) :-).

**Comment by romeric** [comment]
I cannot reproduce this issue with gcc-10 on linux. What error message are you getting? Do you have the latest version of Fastor downloaded. This is related to the other issue #113 which I submitted the fix for yesterday. Are you on macOS or linux?

**Comment by matthiasneuner** [comment]
I am sorry, you were right, I was one commit behind on my notebook. 
Nevertheless, some issues remained (I updated the MWE), which were however **resolved** with yesterday's commit 53c03ed.
Thank you again!
